addons/subscription/models/subscription.py

```diff
@@ -0,0 +1,127 @@
+    doc_lines = fields.One2many('subscription.subscription.history', 'subscription_id', string='Documents created', readonly=True)
+    cron_id = fields.Many2one('ir.cron', string='Cron Job', help="Scheduler which runs on subscription", states={'running':[('readonly',True)], 'done':[('readonly', True)]})
+    note = fields.Text(string='Notes', help="Description or Summary of Subscription")
+
+    @api.model
+    def _auto_end(self):
+        super(SubscriptionSubscription, self)._auto_end()
+        # drop the FK from subscription to ir.cron, as it would cause deadlocks
+        # during cron job execution. When model_copy() tries to write() on the subscription,
+        # it has to wait for an ExclusiveLock on the cron job record, but the latter
+        # is locked by the cron system for the duration of the job!
+        # FIXME: the subscription module should be reviewed to simplify the scheduling process
+        #        and to use a unique cron job for all subscriptions, so that it never needs to
+        #        be updated during its execution.
+        self.env.cr.execute("ALTER TABLE %s DROP CONSTRAINT %s" % (self._table, '%s_cron_id_fkey' % self._table))
+
+    @api.multi
+    def set_process(self):
+        for row in self:
+            mapping = {'name':'name', 'interval_number':'interval_number', 'interval_type':'interval_type', 'exec_init':'numbercall', 'date_init':'nextcall'}
+            res = {'model':'subscription.subscription', 'args': repr([[row['id']]]), 'function':'model_copy', 'priority':6, 'user_id':row['user_id'] and row['user_id'].id}
+            for key,value in mapping.items():
+                res[value] = row[key]
+            cron = self.env['ir.cron'].create(res)
+            self.write({'cron_id':cron.id, 'state':'running'})
+
+    @api.multi
+    def model_copy(self):
+        for row in self:
+            try:
+                id = row['doc_source'].id
+                model_name = row['doc_source']._name
+            except:
+                raise UserError(_('Please provide another source document.\nThis one does not exist!'))
+
+            default = {'state':'draft'}
+            document = self.env['subscription.document'].search([('model.model', '=', model_name)])[0]
+            for f in document.field_ids:
+                if f.value=='date':
+                    value = time.strftime('%Y-%m-%d')
+                else:
+                    value = False
+                default[f.field.name] = value
+
+            state = 'running'
+
+            # if there was only one remaining document to generate
+            # the subscription is over and we mark it as being done
+            if row['cron_id'].numbercall == 1:
+                state = 'done'
+            id = self.env[model_name].copy(id, default)
```

Comment:
Nope, Use recordset(s) to call the method and also change the variable name accordingly .